src/lib/server/endpoints/images.ts

```diff
@@ -0,0 +1,214 @@
+import type { Sharp } from "sharp";
+import sharp from "sharp";
+import type { MessageFile } from "$lib/types/Message";
+import { z } from "zod";
+
+export function createImageProcessorOptionsValidator<MimeType extends string = string>(
+	defaults: ImageProcessorOptions<MimeType>
+) {
+	return z
+		.object({
+			supportedMimeTypes: z
+				.array(
+					z.enum<string, [MimeType, ...MimeType[]]>([
+						defaults.supportedMimeTypes[0],
+						...defaults.supportedMimeTypes.slice(1),
+					])
+				)
+				.default(defaults.supportedMimeTypes),
+			preferredMimeType: z
+				.enum<string, [MimeType, ...MimeType[]]>([
+					defaults.supportedMimeTypes[0],
+					...defaults.supportedMimeTypes.slice(1),
+				])
+				.default(defaults.preferredMimeType),
+			// The 4 / 3 compensates for the 33% increase in size when converting to base64
+			maxSizeInMB: z.number().positive().default(defaults.maxSizeInMB),
+			maxWidth: z.number().int().positive().default(defaults.maxWidth),
+			maxHeight: z.number().int().positive().default(defaults.maxHeight),
+		})
+		.default(defaults);
+}
+
+export interface ImageProcessorOptions<MimeType extends string = string> {
```

Comment:
for the readability, we should put the type/interface defintions after import statements & before definitions of functions/classes